mantidproject/mantid
# Redesign axes tab in figure options to accommodate both 2D and 3D plots

**Description of work.**
This PR changes the axes tab in the figure options so that if the plot is 3D the limits and label can be changed on the z-axis as well as the x and y.

![image](https://user-images.githubusercontent.com/52415735/78258387-1536a880-74f3-11ea-865e-721a96d841fd.png)


**To test:**
Run this script:
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

ws = CreateSampleWorkspace()

fig, ax = plt.subplots(subplot_kw={'projection':'mantid3d'})
ax.plot_surface(ws)

fig.show()
```
Open figure options and in the axes tab you should be able to switch between configuring the x, y, and z axis.
If you change a value for one axis, then switch to another axis and change a value, and then click apply, both changes should apply.
For 3D the axis scale input is disabled as there is an issue with matplotlib where changing the axis scale on a 3D plot doesn't work properly.

Run this script:
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

ws = CreateSampleWorkspace()

fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot(ws, wkspIndex=0)

fig.show()
```
Open figure options and the z-axis option should be disabled.

For good measure, run this script:
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

ws = CreateSampleWorkspace()

fig = plt.figure()

ax1 = fig.add_subplot(121, projection='mantid')
ax1.plot(ws, wkspIndex=0)

ax2 = fig.add_subplot(122, projection='mantid3d')
ax2.plot_surface(ws)

fig.show()
```
And make sure everything in the axes tab works as expected.

Refs #28464 
(doesn't fix because I still need to add a dialog for changing the axis limits when you double-click on the z-axis)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
